### PR TITLE
Adding migrations

### PR DESCRIPTION
**Pull request by btel**
- [x] initialize alembic
- [x] add migration for main tree
- [x] update unit tests to run with migrations
- [x] cli to create followup revision for main tree
- [x] cli to create initial revisions for plugins
- [x] cli to create followup revisions for plugins
- [x] cli to run migrations
- [x] add migrations to all plugin models
- [x] add tests
- [x] CI with/without migrations
- [x] docs
- [x] update cookiecutter mamba-org/quetz-plugin-cookiecutter#3

temporary docs: https://hackmd.io/vScv9P7PTZ2Pcd1dJfDt0w

### CODE REVIEW

File: quetz/database.py [hunk 1]
```diff
@@ -22,7 +20,7 @@ def get_engine(db_url, echo: bool = False, **kwargs) -> Engine:
         kwargs.setdefault('poolclass', StaticPool)
 
     engine = create_engine(db_url, echo=echo, **kwargs)
-    Base.metadata.create_all(engine)
+    # Base.metadata.create_all(engine)
```
Comment (wolfv) [inline quetz/database.py:23]: delete?

Comment (btel) [reply quetz/database.py:23]: that's right :smile:

**Comment by wolfv** [comment]
merge? :)

**Comment by btel** [comment]
@wolfv good to merge

**Comment by wolfv** [comment]
awesome!

**Pull request merged**